File: easy/1637_widest_vertical_area_between_two_points_containing_no_points/solution.cpp
#include <vector>

class Solution {
public:
  int maxWidthOfVerticalArea(std::vector<std::vector<int>> &points) {
    std::sort(points.begin(), points.end());

    int widestGap = 0;

    for (size_t i = 1; i < points.size(); ++i)
      widestGap = std::max(widestGap, points[i][0] - points[i - 1][0]);

    return widestGap;
  }
};

File: easy/1637_widest_vertical_area_between_two_points_containing_no_points/solution.py
from typing import List


class Solution:
    def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:
        points.sort()

        widestGap = 0

        for i in range(len(points) - 1):
            widestGap = max(widestGap, points[i + 1][0] - points[i][0])

        return widestGap
